jdaviz/app.py

```diff
@@ -1017,12 +1017,19 @@ def _viewer_by_reference(self, reference):
 
         Returns
         -------
-        `~glue_jupyter.bqplot.common.BqplotBaseView`
+        viewer : `~glue_jupyter.bqplot.common.BqplotBaseView`
             The viewer class instance.
         """
         viewer_item = self._viewer_item_by_reference(reference)
 
-        return self._viewer_store[viewer_item['id']]
+        # There is a degeneracy between reference and ID.
+        # Imviz falls back to ID for user-created viewers.
+        if viewer_item is not None:
+            viewer = self._viewer_store[viewer_item['id']]
+        else:  # Maybe it is ID
```

Comment:
I used to have this in the app somewhere but maybe it was removed in one of the refactoring. @kecnry , I know your pet peeve is the reference vs ID stuff, so I want to make sure you are okay with this change.